lightning/restore/restore.go

```diff
@@ -660,6 +668,24 @@ func (rc *RestoreController) restoreTables(ctx context.Context) error {
 
 		return errors.New("TiDB Lightning has failed last time; please resolve these errors first")
 	}
+	if len(allDirtyCheckpoints) > 0 {
+		logger := log.L()
+		logger.Error(
+			"TiDB Lightning detects tables with illegal checkpoints. To prevent data mismatch, this run will stop now. Please drop illegal checkpoints first",
+			zap.Int("count", len(allDirtyCheckpoints)),
+		)
+
+		for tableName := range allDirtyCheckpoints {
+			logger.Info("-",
+				zap.String("table", tableName),
+				zap.String("recommendedAction", "./tidb-lightning-ctl --checkpoint-remove='"+tableName+"' --config=..."),
+			)
+		}
+
+		logger.Info("You may also run `./tidb-lightning-ctl --checkpoint-remove=all --config=...` to start from scratch")
+
+		return errors.New("TiDB Lightning detects tables with illegal checkpoints; please remove these checkpoints first")
```

Comment:
```suggestion
		return errors.New("TiDB Lightning has detected tables with illegal checkpoints; please remove these checkpoints first")
```

Reply:
addressed in https://github.com/pingcap/tidb-lightning/pull/317/commits/ef4f3fb78a9737f50f29f3463c87a93cf11f1f1e